### PR TITLE
Enhancement: Build docs for subsets of features/scenarios based on tags

### PR DESCRIPTION
**Pull request by rbcasperson**
### What & Why
A new feature! Hopefully the docs and tests are clear enough to describe the behavior, but here's what it is, and why it can be useful:

This change let's you build documentation for only a subset of your feature files, based on tags. You can provide include tags for only including features/scenarios with those tags, and exclude tags for excluding features/scenarios with those tags.

This is particularly useful for part of my workflow because we want to use sphinx-gherkindoc to show all of the tests from a specific test run. Like if a test run passes, look here to see the coverage of all the behavior that has been validated. So in that case, we would exclude tests that were skipped or expected to fail. sphinx-gherkindoc makes it nice to look at a group of features/scenarios, so it can also make it nice to look at a specific subset of features/scenarios.

### Implementation/Review Notes
The thing that complicated this the most was scenario examples tables. Since examples tables can be tagged, you want to filter those out as well. So an include tag could be on the feature, scenarios, or only the examples table, and that should be included.

There were a bunch of cases I tried to think about. I did a true TDD style where I wrote out all the test cases to describe the behavior expectations and then implemented to make that work. Besides looking at the source code itself, reviewing the test cases for possible missing edge cases will be great. pytest does show that the code coverage for that logic is 100% but that doesn't mean there isn't code that should be there to handle a special case.

The logic is pretty long and possibly confusing (even for me having written it). I tried to refactor as best I could to organize it in the last commit, but I'm definitely happy to get suggestions for how it could be cleaner.

### CODE REVIEW

**Comment by dgou** [comment]
Reviewer note to self: Built this on my work repo and got the same results as with 3.4.6 -- need to look at changes in detail.

File: tests/test_utils.py [hunk 1]
```diff
@@ -281,3 +281,224 @@ def dir_name_to_display_name(dir_name: str):
...
+
+    def __init__(self, tags):
+        self.tags = tags
+
+
+arbitrary_multiple_number = 4
+tagless_scenarios = [
+    Scenario(tags=[], examples=[]) for _ in range(arbitrary_multiple_number)
+]
+feature_tagged_with_a = Feature(scenarios=tagless_scenarios, tags=["a"])
```
Comment (dgou) [inline tests/test_utils.py:315]: :-) This threw me, thinking that there was something missing after `_a` instead of `a` being an arbitrary tag :-)

File: tests/test_utils.py [hunk 2]
```diff
@@ -281,3 +281,224 @@ def dir_name_to_display_name(dir_name: str):
...
+                scenarios=[
+                    Scenario(
+                        tags=[],
+                        examples=examples_tagged_with_a + examples_tagged_with_b,
+                    )
+                ],
+            ),
+            ["a"],
+            ["b"],
+            [Scenario(tags=[], examples=examples_tagged_with_a)],
```
Comment (dgou) [inline tests/test_utils.py:478]: I love the table driven nature of this, but frankly there was a lot of parallelism going on and i just gave up trying to make sure all the hard coded tags and the names containing the tag name(s) were consistent.

File: sphinx_gherkindoc/cli.py [hunk 3]
```diff
@@ -126,7 +126,12 @@ def process_args(
                     integrate_background=args.integrate_background,
                     background_step_format=args.background_step_format,
                     raw_descriptions=args.raw_descriptions,
+                    include_tags=args.include_tags,
+                    exclude_tags=args.exclude_tags,
                 )
+                if not feature_rst_file:
+                    continue
```
Comment (dgou) [inline sphinx_gherkindoc/cli.py:133]: Nothing to see here. Move along...

File: tests/test_utils.py (hunk 1 above)
Comment (rbcasperson) [reply tests/test_utils.py:315]: Ha yeah maybe `x` and `y` would make more sense to avoid that perception. I definitely toyed with how to handle these so I wouldn't have to create a new series of nested features/scenarios/examples.

File: tests/test_utils.py (hunk 2 above)
Comment (rbcasperson) [reply tests/test_utils.py:478]: Honestly I felt like it might be useful to have these tests be written in Gherkin, but somewhat ironically that isn't set up in this repo. The gherkin would do a nice job of pulling out the non-essential/arbitrary values and focus on the difference in variations. The `description` is meant to be the scenario names essentially, so one could use those as the guide for determining if there are any variations missing.

File: sphinx_gherkindoc/writer.py [hunk 4]
```diff
@@ -359,6 +362,11 @@ def table(table: behave.model.Table, inline: bool = False) -> None:
             f" options are: {list(parsers.keys())}"
         )
     feature = feature_class(root_path, source_path)
+
+    included_scenarios = get_all_included_scenarios(feature, include_tags, exclude_tags)
+    if not included_scenarios:
+        return None
```
Comment (sdutton-personal) [inline sphinx_gherkindoc/writer.py:368]: I was concerned about this at first, but I see here that all scenarios are always returned if the tags are not passed:
https://github.com/jolly-good-toolbelt/sphinx_gherkindoc/pull/40/files?file-filters%5B%5D=.py#diff-3d557193ea331bdbcac3270cc1326aafR293

**Comment by sdutton-personal** [comment]
Huh, seems as though checks are still running?  Not sure what the deal is there.

**Comment by rbcasperson** [comment]
Yeah that's weird. I'll take a look.

**Comment by rbcasperson** [comment]
I just reset the last commit hash and repushed and it looks like CI triggered 🤷
